flow/surface_frame.cc

```diff
@@ -26,7 +26,9 @@ SurfaceFrame::SurfaceFrame(sk_sp<SkSurface> surface,
   if (surface_) {
     canvas_ = surface_->getCanvas();
   } else if (display_list_fallback) {
-    dl_recorder_ = sk_make_sp<DisplayListCanvasRecorder>(kGiantRect);
+    FML_DCHECK(!frame_size.isEmpty());
+    dl_recorder_ =
+        sk_make_sp<DisplayListCanvasRecorder>(SkRect::Make(frame_size));
```

Comment:
This is where the magic happens for impeller - now the DL CanvasRecorder can quickReject geometry that falls entirely outside the clip.